visualization/app/codeCharta/util/customViewBuilder.ts

```diff
@@ -0,0 +1,140 @@
+"use strict"
+import {State} from "../codeCharta.model"
+import {CustomView} from "../model/customView/customView.api.model";
+import {CustomViewFileStateConnector} from "../ui/customViews/customViewFileStateConnector";
+
+export class CustomViewBuilder {
+	private static readonly CC_CUSTOM_VIEW_API_VERSION = "1.0.0"
+
+	static buildFromState(viewName: string, state: State): CustomView {
+		const customViewFileStateConnector = new CustomViewFileStateConnector(state.files)
+
+		const customView = {
+			name: viewName,
+			mapSelectionMode: customViewFileStateConnector.getMapSelectionMode(),
+			assignedMap: customViewFileStateConnector.getJointMapName(),
+			mapChecksum: customViewFileStateConnector.getChecksumOfAssignedMaps(),
+			customViewVersion: this.CC_CUSTOM_VIEW_API_VERSION
+		} as CustomView
+
+		customView.stateSettings = {
+			appSettings: undefined,
+			dynamicSettings: undefined,
+			fileSettings: undefined,
+			treeMap: undefined
+		}
+
+		this.initializeAppSettings(customView)
+		this.initializeDynamicSettings(customView)
+		this.initializeFileSettings(customView)
+		this.initializeTreeMapSettings(customView)
+
+		this.deepMapOneToOther(state, customView.stateSettings)
+
+		return customView
+	}
+
+	private static initializeAppSettings(target: CustomView) {
+		target.stateSettings.appSettings = {
+			showMetricLabelNameValue: false,
+			showMetricLabelNodeName: false,
```

Comment:
```suggestion
			showMetricLabelNameValue: true,
			showMetricLabelNodeName: true,
```

from what I understood label should be shown by default, with the option to toggle them off?
Correct me if I'm wrong?

Reply:
Ok I see now that all is set to 0/ false?
Wouldn't it make more sense to set it all to the default settings from the codeCharta.model?